### PR TITLE
Add Kotlin module

### PR DESCRIPTION
**Pull request by filalex77**
Simple Kotlin module, modelled after the `lang/rust` module.

### CODE REVIEW

**Comment by filalex77** [comment]
@hlissner I think I also found out how to do `compile` from project directory. You probably needed that in `lang/rust` :)

**Comment by hlissner** [comment]
This looks ready to merge, except for one small problem: `+kotlin/locate-gradlew-file` uses the naming convention for interactive commands, but is not interactive. Please rename it `+kotlin-locate-gradlew-file`.

File: modules/lang/kotlin/autoload.el [hunk 1]
```diff
@@ -0,0 +1,15 @@
+;;; lang/kotlin/autoload.el -*- lexical-binding: t; -*-
+
+;;;autoload
+(defun +kotlin/locate-gradlew-file ()
```
Comment (hlissner) [inline modules/lang/kotlin/autoload.el]: `+kotlin/locate-gradlew-file` uses the naming convention for interactive commands, but is not interactive. Please rename it to `+kotlin-locate-gradlew-file`.

File: modules/lang/kotlin/doctor.el [hunk 2]
```diff
@@ -0,0 +1,4 @@
+;;; lang/kotlin/doctor.el -*- lexical-binding: t; -*-
+
+(unless (executable-find "ktlint")
+  (warn "ktlint not found. flycheck-kotlin won't work."))
```
Comment (hlissner) [inline modules/lang/kotlin/doctor.el]: Please use the `warn!` macro instead of `warn`.

**Comment by filalex77** [comment]
You're welcome, and thanks for the review!

**Comment by hlissner** [comment]
And done! Thanks again!

**Pull request merged**